datafusion/sql/src/statement.rs

```diff
@@ -425,6 +425,34 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         }))
     }
 
+    fn build_order_by(
+        &self,
+        order_exprs: Vec<OrderByExpr>,
+        schema: &DFSchemaRef,
+    ) -> Result<Vec<datafusion_expr::Expr>> {
+        // Convert each OrderByExpr to a SortExpr:
+        let result = order_exprs
+            .into_iter()
+            .map(|e| self.order_by_to_sort_expr(e, schema))
+            .collect::<Result<Vec<_>>>()?;
+
+        // Verify that columns of all SortExprs exist in the schema:
+        for expr in result.iter() {
```

Comment:
nit: do we really need to loop twice here?

Reply:
Right, I was thinking about we can iterate just once and return resuls if everything is ok or error if something is wrong https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect